o11y/utils.go

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package o11y
+
+import (
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+// ZapDDLogger wraps a ZapSugaredLogger to fit Datadog's logger interface.
+// Required to send the tracer and profiler logs through the SugaredLogger.
+type ZapDDLogger struct {
+	ZapLogger *zap.SugaredLogger
+}
+
+// Log sends an error log through the wrapped zap SugaredLogger
+func (ddLogger ZapDDLogger) Log(msg string) {
+	switch {
+	case strings.Contains(msg, "ERROR"):
```

Comment:
string matching can quickly become inefficient and also prone to cAse-mismatched.

* how does the tracer work? Is it a separate process that will consume our existing logs in a separate thread? 
* does this prevent our existing logs from being published?